src/sourmash/index/sqlite_index.py

```diff
@@ -0,0 +1,1087 @@
+        WHERE {conditions}
+        GROUP BY sourmash_hashes.sketch_id ORDER BY CNT DESC
+        """, template_values)
+
+        return c
+
+
+class SqliteCollectionManifest(BaseCollectionManifest):
+    """
+    A SQLite-based manifest, used both for SqliteIndex and as a standalone
+    manifest class.
+
+    This class serves two purposes:
+    * first, it is a fast, on-disk manifest that can be used in place of
+      CollectionManifest.
+    * second, it can be included within a SqliteIndex (which stores hashes
+      too). In this case, however, new entries must be inserted by SqliteIndex
+      rather than directly in this class.
+
+    In the latter case, the SqliteCollectionManifest is created with
+    managed_by_index set to True.
+    """
+    def __init__(self, conn, *, selection_dict=None, managed_by_index=False):
+        """
+        Here, 'conn' should already be connected and configured.
+
+        Use 'create(filename)' to create a new database.
+
+        Use 'create_from_manifest(filename, manifest) to create a new db
+        from an existing manifest object.
+
+        Use 'load_from_filename' to load from file.
+        """
+        assert conn is not None
+        self.conn = conn
+        self.selection_dict = selection_dict
+        self.managed_by_index = managed_by_index
+        self._num_rows = None
+
+    @classmethod
+    def create(cls, filename):
+        "Connect to 'filename' and create the tables as a standalone manifest."
+        conn = sqlite3.connect(filename)
+        cursor = conn.cursor()
+        cls._create_tables(cursor)
+        return cls(conn)
+
+    @classmethod
+    def load_from_manifest(cls, manifest, *, dbfile=":memory:", append=False):
+        "Create a new sqlite manifest from an existing manifest object."
+        return cls._create_manifest_from_rows(manifest.rows, location=dbfile,
+                                              append=append)
+
+    @classmethod
+    def create_manifest(cls, locations_iter, *, include_signature=False):
+        """Create a manifest from an iterator that yields (ss, location)
+
+        Stores signatures in manifest rows by default.
+
+        Note: do NOT catch exceptions here, so this passes through load excs.
+        Note: this method ignores 'include_signature'.
+        """
+        def rows_iter():
+            for ss, location in locations_iter:
+                row = cls.make_manifest_row(ss, location,
+                                            include_signature=False)
+                yield row
+
+        return cls._create_manifest_from_rows(rows_iter())
+
+    @classmethod
+    def _create_tables(cls, cursor):
+        "Create the manifest table."
+        # this is a class method so that it can be used by SqliteIndex to
+        # create manifest-compatible tables.
+
+        sqlite_utils.add_sourmash_internal(cursor, 'SqliteManifest', '1.0')
+        cursor.execute("""
+        CREATE TABLE sourmash_sketches
```

Comment:
```suggestion
        CREATE TABLE IF NOT EXISTS sourmash_sketches
```

Reply:
Yep, this was intentional; the append case is tested and handled elsewhere, so I wanted an error here!